src/agent/src/main.rs

```diff
@@ -522,6 +514,26 @@ async fn init_attestation_components(logger: &Logger, config: &AgentConfig) -> R
     Ok(())
 }
 
+// Start-up attestation-agent, CDH and api-server-rest if they are packaged in the rootfs
+// and the corresponding procs are enabled in the agent configuration. the process will be
+// launched in the background and the function will return immediately.
+// If the CDH is started, a CDH client will be instantiated and returned.
+async fn init_attestation_components(logger: &Logger, config: &AgentConfig) -> Result<()> {
+    launch_guest_component_procs(logger, config).await?;
+
+    fs::write(OCICRYPT_CONFIG_PATH, OCICRYPT_CONFIG.as_bytes())?;
+    env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", OCICRYPT_CONFIG_PATH);
+
+    // If a CDH socket exists, initialize the CDH client
+    if let Ok(md) = tokio::fs::metadata(CDH_SOCKET).await {
+        if md.file_type().is_socket() {
+            cdh::init_cdh_client(CDH_SOCKET_URI).await?;
+        }
+    }
```

Comment:
It'll be good to capture the error from fs::metadata and log it so that if initialisation fails, we know the reason.
Sorry for additional review comments. I'm figuring out few things while testing.